Add multi-stage scenarios in Access Review API beta (#15387)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/accessreviewinstancedecisionitem-filterbycurrentuser.md

---
title: "accessReviewInstanceDecisionItem: filterByCurrentUser"

> description: "Retrieves all accessReviewInstanceDecisionItem objects on an accessReviewInstance or accessReviewStage for which the calling user is the reviewer."

Reviewer: Same principle of making the description as user-friendly as possible.
```suggestion
description: "Retrieves all decision items on an instance of an access review or a stage of an instance of a multi-stage access review, for which the calling user is the reviewer."
```

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Retrieves all [accessReviewInstanceDecisionItem](../resources/accessreviewinstancedecisionitem.md) objects on a given [accessReviewInstance](../resources/accessreviewinstance.md) or [accessReviewStage](../resources/accessReviewStage.md) for which the calling user is the reviewer.

Reviewer: ```suggestion
Retrieves all decision items on an instance of an access review or a stage of an instance of a multi-stage access review, for which the calling user is the reviewer. The decision items are presented by a [accessReviewInstanceDecisionItem](../resources/accessreviewinstancedecisionitem.md) objects on a given [accessReviewInstance](../resources/accessreviewinstance.md) or [accessReviewStage](../resources/accessReviewStage.md) for which the calling user is the reviewer.
```

> Retrieves all [accessReviewInstanceDecisionItem](../resources/accessreviewinstancedecisionitem.md) objects on a given [accessReviewInstance](../resources/accessreviewinstance.md) for which the calling user is the reviewer.

Reviewer 2: ```suggestion
Retrieves all [accessReviewInstanceDecisionItem](../resources/accessreviewinstancedecisionitem.md) objects on a given [accessReviewInstance](../resources/accessreviewinstance.md) or [accessReviewStage](../resources/accessReviewStage.md) for which the calling user is the reviewer.
```

> If it is a multi-stage access review, retrieves all [accessReviewInstanceDecisionItem](../resources/accessreviewinstancedecisionitem.md) objects on a given [accessReviewStage](../resources/accessReviewStage.md) for which the calling user is the reviewer.

Reviewer 2: ```suggestion
```

Reviewer 2: i think we can just explain this in the examples

GET /identityGovernance/accessReviews/definitions/{accessReviewScheduleDefinitionId}/instances/{accessReviewInstanceId}/decisions/filterByCurrentUser(on='reviewer')
```

Retrieve decisions for a stage in an instance of an access review for which the calling user is the reviewer.
<!-- {
  "blockType": "ignored"
}
-->
``` http
GET /identityGovernance/accessReviews/definitions/{accessReviewScheduleDefinitionId}/instances/{accessReviewInstanceId}/stages/{accessReviewStageId}/decisions/filterByCurrentUser(on='reviewer')
```

> >[!NOTE]

Reviewer: Could this note possibly be in the wrong doc? Coz the scope of this 2nd URL is a particular stageId. Could this note be intended for the first syntax? Or maybe somewhere else?

Reviewer 2: @FaithOmbongi I checked with Kai, and we agree this is in the right place. Per kai, the idea behind this note is to relay that the reviewer can retrieve the decisions from previous stages if the setting is turned on

> >Reviewers can retrieve the decisions from previous stages on the instance if the decisionHistoriesForReviewersEnabled property is enabled in [accessReviewScheduleSettings](../resources/accessreviewschedulesettings.md) on the [accessReviewScheduleDefinition](../resources/accessreviewscheduledefinition.md)

Reviewer 2: ```suggestion
>Reviewers of multi-stage reviews can retrieve the decisions from previous stages on the decsion item if the decisionHistoriesForReviewersEnabled property is enabled in [accessReviewScheduleSettings](../resources/accessreviewschedulesettings.md) on the [accessReviewScheduleDefinition](../resources/accessreviewscheduledefinition.md)
```

            },
            "appliedBy": {
                "id": "00000000-0000-0000-0000-000000000000",
                "displayName": "",
                "userPrincipalName": ""
            },
            "target": {
                "@odata.type": "#microsoft.graph.accessReviewInstanceDecisionItemUserTarget",
                "userId": "1800bb2c-955d-4205-8471-3a6c3116435d",
                "userDisplayName": "guest example",
                "userPrincipalName": "guest@guest.com"
            },
            "principal": {
                "@odata.type": "#microsoft.graph.userIdentity",
                "id": "1800bb2c-955d-4205-8471-3a6c3116435d",
                "displayName": "guest example",
                "userPrincipalName": "guest@contoso.com"
            }
        }
    ]
}
```

<!-- {
  "blockType": "ignored"
}
-->
### Example 2: Retrieves all decisions on an accessReviewStage of a multi-stage access review for which the calling user is the reviewer.

``` http
GET /identityGovernance/accessReviews/definitions/{accessReviewScheduleDefinitionId}/instances/{accessReviewInstanceId}/stages/{accessReviewStageId}/decisions/filterByCurrentUser(on='reviewer')
```

>[!NOTE]

> >reviewer can retrieve the decisions from other stages if the decisionHistoriesForReviewersEnabled is enabled in [accessReviewScheduleSettings](../resources/accessreviewschedulesettings.md)

Reviewer 2: ```suggestion
>Reviewers can retrieve the decisions from previous stages on the instance if the decisionHistoriesForReviewersEnabled property is enabled in [accessReviewScheduleSettings](../resources/accessreviewschedulesettings.md) on the [accessReviewScheduleDefinition](../resources/accessreviewscheduledefinition.md)
```

            "reviewedBy": {
                "id": "36c4c56e-fce3-4e2d-b28e-4ac0c7d2fa10",
                "displayName": "MOD Administrator",
                "userPrincipalName": "MOD Administrator"
            },
            "appliedBy": {
                "id": "00000000-0000-0000-0000-000000000000",
                "displayName": "",
                "userPrincipalName": ""
            },
            "target": {
                "@odata.type": "#microsoft.graph.accessReviewInstanceDecisionItemUserTarget",
                "userId": "1800bb2c-955d-4205-8471-3a6c3116435d",
                "userDisplayName": "guest example",
                "userPrincipalName": "guest@guest.com"
            },
            "principal": {
                "@odata.type": "#microsoft.graph.userIdentity",
                "id": "1800bb2c-955d-4205-8471-3a6c3116435d",
                "displayName": "guest example",
                "userPrincipalName": "guest@contoso.com"
            }
        }
    ]
}
```

### Example 2: Retrieves all decisions on an accessReviewStage of a multi-stage access review for which the calling user is the reviewer.

#### Request

<!-- {
  "blockType": "request",

>   "name": "accessreviewinstancedecisionitem_filterbycurrentuser"

Reviewer: Please make this a unique name. Helps avoid clashes in code snippet names and the wrong snippet being appended to an example.

>   "name": "accessreviewinstancedecisionitem_filterbycurrentuser"

Reviewer 2: ```suggestion
  "name": "accessreviewinstancedecisionitem_filterbycurrentuser_2"
```
